octodns/provider/ns1.py

```diff
@@ -304,7 +332,9 @@ def _data_for_geo_A(self, _type, record):
 
     def _data_for_dynamic_A(self, _type, record):
         # First make sure we have the expected filters config
-        if self._DYNAMIC_FILTERS != record['filters']:
+        unsupported_fcs = False in [fc in self._SUPPORTED_FILTER_CONFIGS for
+                                    fc in record['filters']]
+        if not record['filters'] or unsupported_fcs:
```

Comment:
If I'm reading this correctly I believe it won't take order into consideration which we need to. The ordering of the filters is important for octoDNS's dynamic model.

Reply:
Yup, didn't realize the equality check that I replaced was also checking the order. Will address.